concordium-consensus/src/Concordium/KonsensusV1/Types.hs

```diff
@@ -182,6 +186,27 @@ finalizerByIndex :: FinalizationCommittee -> FinalizerIndex -> Maybe FinalizerIn
 finalizerByIndex finCom finInd =
     committeeFinalizers finCom Vector.!? fromIntegral (theFinalizerIndex finInd)
 
+-- | Hash of the finalization committee, only the weight and BLS verify key of each finalizer are used
+-- for computing this.
+newtype FinalizationCommitteeHash = FinalizationCommitteeHash
+    { theFinalizationCommitteeHash :: Hash.Hash
+    }
+
+-- | Compute the hash of the finalization committee. Only the weight and BLS verify key of each
+-- finalizer are used for computing this.
+computeFinalizationCommitteeHash :: FinalizationCommittee -> FinalizationCommitteeHash
+computeFinalizationCommitteeHash FinalizationCommittee{..} =
+    FinalizationCommitteeHash $
+        LFMBT.hashAsLFMBT emptyCommitteeHash $
+            computeFinalizerInfoHash <$> Vector.toList committeeFinalizers
+  where
+    emptyCommitteeHash = Hash.hash "EmptyFinalizationCommittee"
+    computeFinalizerInfoHash :: FinalizerInfo -> Hash.Hash
+    computeFinalizerInfoHash info = Hash.hashOfHashes weightHash blsKeyHash
+      where
+        weightHash = Hash.hash $ runPut $ put (finalizerWeight info)
+        blsKeyHash = Hash.hash $ runPut $ put (finalizerBlsKey info)
```

Comment:
The serialization of the BLS key is quite expensive typically.

How often will this be serialized/computed? If it is often we might consider storing a hash of the key since it changes very rarely, and not more than once a day for the committee.

Reply:
Why not just cache a hash of the whole finalization committee? The finalization committee itself does not change in between paydays.

Reply:
> How often will this be serialized/computed?

I did not expect this to be expensive, but to answer the question, this will be computed every time a block is baked or processed by another node.

> Why not just cache the hash of the whole finalization committee?

I did consider this as well during development, but did not consider this computation expensive enough to be worth it.
I have now changed the behavior to be caching it, but put the cache as part of the type `BakersAndFinalizers` instead, since the `FinalizationCommittee` type seems to be used as part of timeout network messages.